src/app/views/sidebar/sample-queries/SampleQueries.tsx

```diff
@@ -329,7 +343,9 @@ export class SampleQueries extends Component<ISampleQueriesProps, any> {
 
     const { sampleQueries } = this.state;
     const classes = classNames(this.props);
-    const groups = generateGroupsFromList(sampleQueries, 'category');
+    if (this.resetCollapse) {
+      this.groups = generateGroupsFromList(sampleQueries, 'category');
+    }
```

Comment:
Since the category is already known in the `selectedQuery` state property, it will be pretty easy to get the category and make it expanded

```suggestion
    const groups = generateGroupsFromList(sampleQueries, 'category');
    if (this.state.selectedQuery) {
      const index = groups.findIndex(k => k.key === this.state.selectedQuery.category);
      groups[index].isCollapsed = false;
    }
```